src/main/java/address/Browser/BrowserManager.java

```diff
@@ -1,53 +1,177 @@
+        int indexOfContactsInFilteredListView = filteredPersons.size();
+
+        int indexOfSelectedTabBrowserPage = findIndexOfTabBrowserThatContainsPersonGitHubWebPage(person);
+
+        if (indexOfSelectedTabBrowserPage == PERSON_DOES_NOT_EXISTS_IN_BROWSER) {
+            indexOfSelectedTabBrowserPage = findASuitableBrowserTabAndLoadThePersonGitHubProfilePage(
+                                            person, filteredPersons, indexOfContactsInFilteredListView);
+        }
+        browser.selectTab(indexOfSelectedTabBrowserPage);
+
+        updateOtherBrowserTabs(filteredPersons, indexOfSelectedTabBrowserPage);
+    }
+
+    /**
+     * Returns the index of the tab in the TabBrowser that contains the person's GitHub profile page.
+     * @param person
+     * @return the index of the tab in the TabBrowser that contains the person's GitHub profile page
+     *         or -1 if the tabs of the TabBrowser do not contain the person's GitHub profile page.
+     */
+    private int findIndexOfTabBrowserThatContainsPersonGitHubWebPage(Person person) {
+        int indexOfSelectedTabBrowserPage = PERSON_DOES_NOT_EXISTS_IN_BROWSER;
+        for (TabBrowserPage page : browser.getAllPages()) {
+            if (!page.isAssignedToLoadAPersonGitHubProfilePage()) {
+                continue;
+            }
+            if (page.getPerson().equals(person)) {
+                indexOfSelectedTabBrowserPage = browser.indexOf(page);
+                break;
+            }
+        }
+        return indexOfSelectedTabBrowserPage;
     }
 
-    private void automateClickingAndScrolling() {
-        DOMElement container = browser.getDocument().findElement(By.id("js-pjax-container"));
-        DOMElement link = browser.getDocument().findElement(By.className("octicon octicon-repo"));
-        if (link != null) {
-            container.addEventListener(DOMEventType.OnLoad, e ->
+    private int findASuitableBrowserTabAndLoadThePersonGitHubProfilePage(Person person, List<Person> persons,
+                                                                         int indexOfContactsInFilteredListView) {
+
+        int indexOfSelectedTabBrowserPage = 0;
+
+        for (TabBrowserPage page : browser.getAllPages()) {
+            if (!page.isAssignedToLoadAPersonGitHubProfilePage()) {
+                continue;
+            }
+            int pagePersonIndex = persons.indexOf(page.getPerson());
+            if (pagePersonIndex < indexOfContactsInFilteredListView) {
+                indexOfContactsInFilteredListView = pagePersonIndex;
+                indexOfSelectedTabBrowserPage = browser.indexOf(page);
+            }
+        }
+        browser.getAllPages().get(indexOfSelectedTabBrowserPage).loadURL(person.getPersonGitHubProfilePageUrl(),
+                person);
+        return indexOfSelectedTabBrowserPage;
+    }
+
+    private void updateOtherBrowserTabs(List<Person> persons, int indexOfSelectedTabBrowserPage) {
```

Comment:
Comment on what kind of updates etc.

Use standard javadoc header comments, especially elaborate on the `@param`s and `@return`s to clarify the input and output
```java
/**
 * <description>
 * @param persons <purpose and constraints>
 * @param selectedTabIndex <purpose>
 */
```